client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-plan-features-for-grid-plans.ts

```diff
@@ -85,6 +88,21 @@ const usePlanFeaturesForGridPlans: UsePlanFeaturesForGridPlans = ( {
 			};
 		} );
 
+		if ( highlightedFeatures && highlightedFeatures.length > 0 ) {
+			highlightedFeatures.reverse().forEach( ( slug ) => {
+				const feature = wpcomFeatures.find( ( feature ) => feature.getSlug() === slug );
+				if ( feature ) {
+					const availableOnlyForAnnualPlans = annualPlansOnlyFeatures.includes( feature.getSlug() );
+					wpcomFeaturesTransformed.unshift( {
+						...feature,
+						availableOnlyForAnnualPlans,
+						availableForCurrentPlan: ! isMonthlyPlan || ! availableOnlyForAnnualPlans,
+						isHighlighted: true,
+					} );
+				}
+			} );
+		}
+
```

Comment:
The use of unshifting here, which I duplicated from how topFeature is handled just below, seems non-ideal. It changes the order - I needed to add .reverse() to try to keep same order as features are defined in plans-list file.

Reply:
I think the entire logic here could be refactored/abstracted a little better, into a separate hook maybe. "entire logic" including the top-feature computations. So the process could be like:

1. parse the features and transform
2. send the transformed features through a function to "reorder" and "highlight" (and grab the highlighted set and selected feature there)

but it needs a deeper refactor. something for later maybe

Reply:
Also, we are apparently just following along how it's done for the top-feature here, but that also means we are restricting this to WPCOM features, so Jetpack features cannot be highlighted. We'd need to revisit if it's something we intend to do down the line.

I don't mean to drag this PR longer. I would have considered maybe a better abstraction to include both sets from the get-go to be complete, but totally fine to leave it for later. I can spin off a separate issue to revisit

Reply:
Yes, it may be worth refactoring/simplifying all the code in this hook that reorders and transforms the features (including highlightedFeatures, topFeatures, annualPlansOnlyFeatures). Also agree it may be better handle that in a separate PR. The map() that transforms the jetpack features is much nicer and simpler, in large part because it doesn't need to do so much conditional re-ordering first.